infra/aws/main.tf

```diff
@@ -0,0 +1,163 @@
+resource "buildkite_agent_token" "tf_managed" {
+  description = "token used by the build fleet"
+}
+
+resource "aws_ssm_parameter" "bk_agent_token" {
+  name  = "/bk_agent_token"
+  type  = "String"
+  value = buildkite_agent_token.tf_managed.token
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "5.0.0"
+
+  name = "vllm-ci-vpc"
+  cidr = "10.0.0.0/16"
+
+  azs            = ["us-west-2a", "us-west-2b", "us-west-2c", "us-west-2d"]
+  public_subnets = ["10.0.1.0/24", "10.0.2.0/24", "10.0.3.0/24", "10.0.4.0/24"]
+
+  enable_dns_hostnames          = true
+  map_public_ip_on_launch       = true
+  manage_default_network_acl    = false
+  manage_default_route_table    = false
+  manage_default_security_group = false
+
+  tags = {
+    Name = "vLLM CI VPC"
+  }
+}
+
+locals {
+  default_parameters = {
+    elastic_ci_stack_version = var.elastic_ci_stack_version
+
+    BuildkiteAgentTokenParameterStorePath = aws_ssm_parameter.bk_agent_token.name
+    MinSize                               = 0
+    EnableECRPlugin                       = "true"
+    VpcId                                 = module.vpc.vpc_id
+    SecurityGroupIds                      = module.vpc.default_security_group_id
+    Subnets                               = join(",", module.vpc.public_subnets)
+    RootVolumeSize                        = 512   # Gb
+    EnableDockerUserNamespaceRemap        = false # Turn off remap so we can run dind
+    BuildkiteAgentTimestampLines          = true
+    BuildkiteTerminateInstanceAfterJob    = true
+  }
+
+  queues_parameters = {
+    cpu-queue = {
+      BuildkiteQueue          = "cpu_queue"
+      InstanceTypes           = "r6in.16xlarge"
+      MaxSize                 = 2
+      ECRAccessPolicy         = "poweruser"
+      InstanceOperatingSystem = "linux"
+      OnDemandPercentage      = 100
+      EnableInstanceStorage   = "true"
+    },
+
+    gpu-1-queue = {
+      BuildkiteQueue          = "gpu_1_queue"
+      InstanceTypes           = "g6.4xlarge"
```

Comment:
why `1-queue` is 4x large and `4-queue` is 12xlarge? it the number in the queue name more like the number of gpus? could you add a comment that explains?

Reply:
done